spec/models/classrooms_teacher_spec.rb

```diff
@@ -21,40 +38,26 @@
     end
   end
 
-  # describe 'callbacks' do
-  #   let(:classroom) { create(:classroom, :with_no_teacher) }
-  #   let(:teacher) { create(:teacher) }
-  #   let(:classrooms_teacher) { build(:classrooms_teacher,
-  #     classroom_id: classroom.id,
-  #     user_id: teacher.id
-  #   )}
-  #   # it 'should trigger_analytics_events_for_classroom_creation on create commit' do
-  #   #   current_jobs = ClassroomCreationWorker.jobs.size
-  #   #   classrooms_teacher.save!
-  #   #   expect(ClassroomCreationWorker.jobs.size).to eq(current_jobs + 1)
-  #   # end
-  #
-  #   it 'should delete_classroom_minis_cache on create' do
-  #     $redis.set("user_id:#{teacher.id}_classroom_minis", {something: 'something'})
-  #     classrooms_teacher.save
-  #     expect($redis.get("user_id:#{teacher.id}_classroom_minis")).to eq nil
-  #   end
-  # end
-
-  describe 'associations' do
-    let(:classroom) { create(:classroom, :with_no_teacher) }
-    let(:teacher) { create(:teacher) }
-    let(:classrooms_teacher) { create(:classrooms_teacher,
-      classroom_id: classroom.id,
-      user_id: teacher.id
-    )}
-
-    it 'should get the right teacher' do
-      expect(classrooms_teacher.teacher).to eq(teacher)
+  describe 'callbacks' do
+    let(:classrooms_teacher) { build(:classrooms_teacher) }
+    let(:teacher) { classrooms_teacher.teacher }
+
+    # these specs fail because self.owner is called in the callback which is not defined
```

Comment:
This should work properly now -- I changed `self.owner` to `self.classroom.owner` in the `trigger_analytics_events_for_classroom_creation` method. Could you verify it works and then remove this comment if so?

Reply:
I saw your commits, I forgot to remove it earlier, I've removed it now